DEV-GUIDE.md

```diff
@@ -367,6 +369,135 @@ Please refer to the
+
+Our reference documentation comes from two places:
+
+- Comments on proto files for our config formats and API definitions. The various
+protos in the `istio/api` repo, the Mixer adapter configuration protos, and the Mixer
+template definitions are all examples of this.
+
+- Cobra commands for our CLI docs. The various CLI commands we build generally use the
+Cobra framework to parse our command-lines. Cobra is also used to produce 
+reference documentation for the individual commands.
+
+Here's how this works:
+
+- Developers comment every element in proto files they author
+
+- Developers provide quality help & usage text in Cobra command definitions.
+
+- Whenever we compile protos, in addition to producing `.pb.go` files, our build system
+also produces `.pb.html` files which hold the documentation for the particular proto package.
+
+- Within the `istio/istio.github.io` repo, the script file `scripts/grab_reference_docs.sh` 
+does the work necessary to harvest all the generated `.pb.html` files from the `istio/api`
+and `istio/istio` repos and installs them in the right place in the website hierarchy.
+The script also builds and runs the various CLI commands we have in order to extract their
+documentation, and copies their docs to the right place in the website hierarchy. Once the
+script is done running, the changes to the website just need to be pushed and istio.io will
+shortly thereafter reflect the changes.
+
+### About proto documentation
+
+Writing proto documentation is a simple matter of adding comments to elements
+within the input proto files. You can put comments directly above individual elements, or to the
+right. For example:
+
+```proto
+// A package-level comment
+package pkg;
+
+// This documents the message as a whole
+message MyMsg {
+    // This documents this field 
+    // It can contain many lines.
+    int32 field1 = 1;
+
+    int32 field2 = 2;       // This documents field2
+}
+```
+
+Comments are treated as markdown. You can thus embed classic markdown annotations within any comment.
+
+#### Linking to types and elements
+
+In addition to normal markdown links, you can also use special proto links within any comment. Proto
+links are used to create a link to other types or elements within the set of protos. You specify proto links
+using two pairs of square brackets such as:
+
+```proto
+// This is a comment that links to another type: [MyOtherType][MyPkg.MyOtherType]
+message MyMsg {
+
+}
+
+```
+
+The first square brackets contain the name of the type to display in the resulting documentation. The second
+square brackets contain the fully qualified name of the type or element being referenced, including the
+package name.
+
+#### Annotations
+
+Within a proto file, you can insert special comments which provide additional metadata to
+use in producing quality documentation. Within a package, include an unattached
+comment of the form:
+
+```
+// $title: My Title
+// $overview: My Overview
```

Comment:
What is the difference in rendering between package level comment and $overview ? Is it like long and short description ?

Reply:
If package level comment is required can you mention that too.

Consider : it will be helpful to bold **must** for things like exactly 1 fine in the package **must** contain $title, $overview and $location and the same file **must** also contain package level comment.